client/files.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+)
+
+// MakeDirOptions holds the options for a call to MakeDir.
+type MakeDirOptions struct {
+	// Path is the absolute path of the directory to be created (required).
+	Path string
+
+	// MakeParents specifies whether the non-existing parent directories will
+	// be created or not. If one of the parent directories of the specified
+	// path does not exist and MakeParents is false, the call will fail.
+
+	// MakeParents, if true, specifies that any non-existent parent directories
+	// should be created. If false (the default), the call will fail if the
+	// directory to be created has at least one parent directory that does not
+	// exist.
+	MakeParents bool
+
+	// Permissions specifies the permission bits of the directories to be created.
```

Comment:
We should probably document that the directory is created with permission 0o755 if this is nil.

Alternatively, it might simplify the API to make this a non-pointer, with 0 meaning "use default". It seems to me it's not useful to use 0 as a permission value.